simpleble/include/simpleble/Adapter.h

```diff
@@ -42,11 +52,17 @@ class SIMPLEBLE_EXPORT Adapter {
     static bool bluetooth_enabled();
 
     /**
-     *  Fetches a list of all available adapters.
+     * Fetches a list of all available adapters from all available backends.
+     *
+     * This will cause backends to be instantiated/initialized and adapters
+     * too.
      */
     static std::vector<Adapter> get_adapters();
 
   protected:
+    AdapterBase* operator->();
+    const AdapterBase* operator->() const;
+
     std::shared_ptr<AdapterBase> internal_;
```

Comment:
@kdewald This repetition in each frontend class could be avoided by having a base PIMPL class that is (privately) inherited. I did not do it not to make the header look "complicated" but it would save on code.

Reply:
I'm fine having the duplication at this point, the most important thing is that the user-facing API is easy to read and understand right away. The extra lines of code are negligible.